geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -2924,13 +2809,13 @@ private void addLocalHostAsServer(PoolFactory poolFactory) {
   }
 
   /**
-   * Used to set the default pool on a new GemFireCache.
+   * Set the default pool on a new GemFireCache.
    */
-  @Override
-  public synchronized void determineDefaultPool() {
+  private synchronized void determineDefaultPool() {
     if (!isClient()) {
-      throw new UnsupportedOperationException();
+      throw new UnsupportedOperationException("determineDefaultPool requires a client cache");
```

Comment:
TODO: Wording should be improved to not use the name of a private method.

Reply:
Is it possible (and reasonable) to move this check up the call stack, so that the exception message can refer to more public methods?

Reply:
I'll look into that.